### ISSUE TITLE
Drag tab to open new window

### ISSUE THREAD

**Issue by wgordon17** [zed-industries/zed#6722]
### Check for existing issues

- [X] Completed

### Describe the feature

I'm torn if this is a bug vs an RFE, but I definitely find this an _important_ functionality. Working with multiple monitors, I find being very productive utilizing multiple editor windows. I know that Zed well supports split-windows within the same monitors, but I don't _prefer_ that method as my screens are a little on the small side so that I can support have 4 external monitors.

The expected behavior would be that I can drag an open tab to another monitor and it would spawn a new editor window with the tab.

**Comment by mpiannucci** [comment]
This is probably the best feature that i remember from sublime text. VSCode and Zed cant do it and it drives me crazy. I would love this in Zed

**Comment by paul-hansen** [comment]
>  VSCode and Zed cant do it and it drives me crazy.

@mpiannucci Just a heads up, VSCode can do this now. It didn't used to so it might just be you haven't tried it since they added it. See [this comment](https://github.com/microsoft/vscode/issues/10121#issuecomment-1836212530) for a gif of how it works.

**Comment by notpeter** [comment]
VSCode supports `Move to new Window` and `Copy to New Window cmd+k o` when right clicking on tabs too.
<img width="329" alt="image" src="https://github.com/zed-industries/zed/assets/145113/e3026f0f-5b36-49ee-85c0-5ea5abeb7ee5">

**Comment by Roeegg2** [comment]
late to the party, but anything has been done about this? it is really frustrating (and unlike VScode Zed won't allow me to open the same projects in 2 windows, so i can't do that too)

**Comment by ion-elgreco** [comment]
I am a bit surprised this is not a feature, apparently I used this quite often in a multi-monitor setup with VSCode

**Comment by gdvine** [comment]
+1, great feature from Sublime Text for basic workflow

**Comment by zophiale** [comment]
I'm gonna try looking at this and coding it in.
wish me luck lmao
(spite post)

**Comment by svechinsky** [comment]
Anyone have any idea what's the ETA on this, it's a blocker to quite a few workflows imo

**Comment by matteomonti** [comment]
> Anyone have any idea what's the ETA on this, it's a blocker to quite a few workflows imo

Absolutely! Zed seems amazing, but not being able to work on more than one monitor for now is a complete deal breaker for me! I look forward to the project growing though, it really looks super cool!

**Comment by aretrace** [comment]
I'm a bit surprised this isn't supported, sad 😔

**Comment by ddoemonn** [comment]
Hey, I'd like to work on this if it's okay with the team. Happy to discuss the approach before diving in.

**Comment by wolfhive** [comment]
Multi-window support (one workspace) is a must-have for tile window managment in Linux (Niri, Hyprland, sway)

**Comment by hcb12311** [comment]
@marcocondrache non-technical here - managing multiple agents working on different shit - this is essential. cant believe this doesnt exist yet. i have 8 dif agents from codex/CC working on 8 dif parts of the codebase and cant see a single fucken one of them xD
thanks boss

**Comment by esthertrapadoux** [comment]
@marcocondrache assigned this to you!!

**Comment by jamest-ultra** [comment]
Hey, just wondering if there are any updates to this. It would be the absolute final thing to fully convert me over to Zed.

**Comment by tetrahedron-systems** [comment]
yeah, how do I move a tab into a new window?

**Comment by musel25** [comment]
im really waiting for this!

**Comment by fritzjung** [comment]
maaaaaaan, two years this has been brewing.

**Comment by CalvinWilkinson** [comment]
> Yeah, [#44489](https://github.com/zed-industries/zed/pull/44489) does clone the workspace, but I don't think that's the approach we want. From a performance point of view, this would presumably mean running multiple copies of LSPs, which for me is already a bit of a non-starter (a single rust-analyzer instance in the Zed codebase uses >10GB of RAM). If you're having success with the satellite approach, I'd be very keen to see that merged.
> 
> AFAICT, your PR doesn't implement dragging, which I think is an important UX feature, but of course (assuming [#47704](https://github.com/zed-industries/zed/pull/47704) gets merged), [#44489](https://github.com/zed-industries/zed/pull/44489) could rebase and use your satellite implementation. I'll assign myself to [#47704](https://github.com/zed-industries/zed/pull/47704) and review it today. Thanks 😁

100%.  The tab being in a new window should not be a cloned workspace, and I agree for the reasons stated above.  Making this editor feature parity with VSCode, with the major features that matter, I think, would be a huge win for this editor.  I am loving this editor so far and you guys have something special here.  Keep up the great work!

**Comment by CalvinWilkinson** [comment]
> Hey, just wondering if there are any updates to this. It would be the absolute final thing to fully convert me over to Zed.

I agree.  This feature would make me switch over for good and drop VSCode.

**Comment by CalvinWilkinson** [comment]
I am curious if this is being worked on at all and what the priority is.  This is a very important feature for many people and would amazing to get.  I noticed the pr #44489 has been closed and not merged.  It seems like it was working, but no explanation why it is not being merged in.

**Comment by Niach** [comment]
Im working on a similar project using gpui under the hood and having the necessary changes for this feature in gpui core would be a game changer!

**Comment by axeldvp** [comment]
This feature would be awesome !

**Comment by Waltarb** [comment]
We got to have this I cant even open the project in a second window so you are stuck with one panel. Got me moving back to vscode honestly.
